core/src/main/java/com/microsoft/applicationinsights/telemetry/JsonTelemetryDataSerializer.java

```diff
@@ -320,35 +335,76 @@ private void writeName(String name) throws IOException {
         return ret;
     }
 
-    protected void writeEscapedString(String value) throws IOException {
-        for (char c : value.toCharArray()) {
-            switch (c) {
-                case '\\':
-                    out.write("\\\\");
-                    break;
-                case '"':
-                    out.write("\\\"");
-                    break;
-                case '\n':
-                    out.write("\\n");
-                    break;
-                case '\b':
-                    out.write("\\b");
-                    break;
-                case '\f':
-                    out.write("\\f");
-                    break;
-                case '\r':
-                    out.write("\\r");
-                    break;
-                case '\t':
-                    out.write("\\t");
-                    break;
-                default:
-                    out.write(c);
+    private String sanitizeStringForJSON(String text, int maxLength) {
+
+        final StringBuilder result = new StringBuilder();
+        StringCharacterIterator iterator = new StringCharacterIterator(text);
+        for (char curr = iterator.current(); curr != iterator.DONE && result.length() < maxLength - 2; curr = iterator.next()) {
```

Comment:
Minor: 2 looks like a magic number here.

Reply:
Yes 2 is a magic number. This is to ensure safety because you need that two additional character space if the last letter is an escape letter. Any better ways to address this?

Reply:
You can make it constant with the name explaining what it is.

Reply:
Yes I can make it constant. Will change it.